### PR TITLE
[WIP DO NOT SUBMIT] Enable LTO on Darwin platforms.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This save about 1.6 megs from the final Flutter dylib in release mode (13.1 megs to 11.5). These are uncompressed sizes.

I tried enabling LTO on Android but we wind up using the NDK linker on the same and `-flto` to Clang makes it generate LLVM bitcode (also why I switched to `llvm-ar` on the `alink` tool in this patch).

Makes link times go way up so maybe we should only enable this in some variants.

### CODE REVIEW

File: build/toolchain/mac/BUILD.gn [hunk 1]
```diff
@@ -227,8 +228,8 @@ mac_toolchain("ios_clang_x64") {
   toolchain_cpu = "x64"
   toolchain_os = "mac"
   prefix = rebase_path(llvm_bin_path, root_build_dir)
-  cc = "${goma_prefix}$prefix/clang"
-  cxx = "${goma_prefix}$prefix/clang++"
+  cc = "${goma_prefix}$prefix/clang -flto"
+  cxx = "${goma_prefix}$prefix/clang++ -flto"
```
Comment (abarth) [inline build/toolchain/mac/BUILD.gn]: Shouldn't this flag be in cflags?

**Comment by abarth** [comment]
Other projects I've worked on have had a "production" build configuration that enabled flags like LTO which are useful in production but make development super painful.

**Comment by eseidelGoogle** [comment]
FYI @a-siva

**Comment by chinmaygarde** [comment]
I am going to clean up this patch and submit another one.

**Pull request closed without merging**